lib/takeaway.rb

```diff
@@ -0,0 +1,34 @@
+require_relative '../lib/menu.rb'
+require_relative '../lib/basket.rb'
+require_relative '../lib/text.rb'
+
+class Takeaway
+
+  def initialize(menu = Menu.new, current_basket = Basket.new, sms = Text.new)
+    @menu = menu
+    @current_basket = current_basket
+    @sms = sms
+  end
+
+  def menu
+    @menu.view
+  end
+
+  def order(dish)
+    fail "Dish unavailable" unless @menu.list.has_key? dish
+    cost = @menu.list[dish]
+    @current_basket.basket[dish] = cost
```

Comment:
The 3 lines above could have been moved the `basket` class into a new `add` method as this is functionality `Basket` should be responsible

Reply:
Great feedback here :+1: